inc/class-s3-uploads.php

```diff
@@ -208,12 +209,108 @@ public function filter_editors( $editors ) {
 	 */
 	public function filter_sideload_move_temp_file_to_s3( array $file ) {
 		$upload_dir = wp_upload_dir();
-		$new_path = $upload_dir['basedir'] . '/tmp/' . basename( $file['tmp_name'] );
+		$new_path   = $upload_dir['basedir'] . '/tmp/' . basename( $file['tmp_name'] );
 
 		copy( $file['tmp_name'], $new_path );
 		unlink( $file['tmp_name'] );
 		$file['tmp_name'] = $new_path;
 
 		return $file;
 	}
+
+	/**
+	 * Filters wp_read_image_metadata. exif_read_data() doesn't work on
+	 * file streams so we need to make a temporary local copy to extract
+	 * exif data from.
+	 *
+	 * @param array  $meta Array of metadata
+	 * @param string $file File name on S3
+	 * @return array
+	 */
+	public function wp_filter_read_image_metadata( $meta, $file ) {
```

Comment:
Rather than copy the entire function, would we move the filter to our own function and call it with the copied file:

```PHP
remove_filter( ...the old filter );
add_filter( filter, array( $this, 'replaced_function' ) );

function replaced_function( $file ) {
    $new_file = copy_file_from_s3( $file );
    $data = old_function( $new_file );
    unlink( $file );
    return $data;
}
```